cpu/sam3x8e/periph/pwm.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (C) 2015 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_sam3x8e
+ * @{
+ *
+ * @file
+ * @brief       CPU specific low-level PWM driver implementation for the SAM3X8E
+ *
+ * @author      Andreas "Paul" Pauli" <andreas.pauli@haw-hamburg.de>
+ *
+ * @}
+ */
+#include <stdint.h>
+#include "board.h"          /* -> cpu.h ??? */
+#include "periph/pwm.h"     /* "driver", interface of pwm.c ONLY */
+
+/*
+ * guard file in case no PWM device is defined, all #define'd in "periph_conf.h"
+ */
+#if PWM_NUMOF && (PWM_0_EN || PWM_1_EN || PWM_2_EN || PWM_3_EN)
+
+#define ERR_INIT_MODE           (-1)
+#define ERR_INIT_BWTH           (-2)
+#define ERR_SET_CHAN            (-1)
+
+// Disable IO on PC2 and set Peripheral device B (PWM0L)
+#define PWM_0_PIN_SET                           \
+    PIOC->PIO_PDR |= PIO_PDR_P2;                \
+    PIOC->PIO_ABSR |= PWM_0_PIN;
+
+/*!
```

Comment:
I am unsure if this documentation should be part of doxygen. Any other opinions here? Anyway, please clarify the ToDos and remove them from the doc